### ISSUE TITLE
Consistent callbacks

### ISSUE THREAD

**Issue by Bloke** [textpattern/textpattern#964]
Over the years, the Txp callback system has expanded in a slightly haphazard manner. Perhaps it's time to take stock of what we have and migrate callbacks to a more consistent framework?

[Background to the request](https://forum.textpattern.io/viewtopic.php?id=48126).

Changing callback names is problematic because of the number of legacy plugins that hook into them. Could we investigate a [migration path](https://forum.textpattern.io/viewtopic.php?pid=307539#p307539) perhaps? Specifically, is it possible to introduce parallel callbacks (possibly via an array mechanism in the callback_event(_ref) functions?) so that both old and new callbacks could co-exist for a few versions until we gradually phase out the old ones?

@makss @bloatware What do you think? Is it possible / desirable and can it be done without much slowdown or backwards compatibility breakage? Or are there any better ideas? Maybe introduce a new OO callback registry entirely that would use the new callback names and support the old events/steps as 'aliases' for now? Thus the two could co-exist cleanly.

If this has potential, we should first of all define an extensible callback scheme for the various places we use it, such as front-end, API, and panels (including the problematic Images panel). Thoughts welcome.

**Comment by NicolasGraph** [comment]
Any news about this I could use to edit in the _Skin_ package related callback events before merging?

**Comment by makss** [comment]
Yes, we need to bring order with callbacks. I'm not ready to think about it yet.

Attempt to automatically generate the actual callback list:
`npm run txp-index`

This will create several html files in the temporary directory `.github/index/`.
